mlir/include/mlir/Dialect/MLProgram/Transforms/Passes.td

```diff
@@ -11,6 +11,14 @@
 
 include "mlir/Pass/PassBase.td"
 
+def MLProgramBufferize: Pass<"mlprogram-bufferize", "ModuleOp"> {
+  let summary = "Bufferize the MLProgram dialect ops";
```

Comment:
Please provide an extensive documentation here.

It's not clear to me why we need special pass here instead of reusing the one-shot buffering, which I would expect to work with interfaces.

Reply:
Added doc. One-Shot Bufferize does not visit module level ops like `ml_program.global` so implementing the `BufferizableOpInterface` did not work (see my comment for Matthias below).

Reply:
Can we instead improve/fix the bufferization pass?